python/cudf/cudf/core/dataframe.py

```diff
@@ -3200,6 +3207,8 @@ def insert(self, loc, name, value, nan_as_null=None):
             ``null`` values.
             If ``False``, leaves ``np.nan`` values as is.
         """
+        if nan_as_null is no_default:
```

Comment:
Should this be in `_insert` instead?

Reply:
I haven't touched `_insert` and other private APIs like column constructor because I felt having this handling in public APIs will just be sufficient.